app/frontend/components/shared/permit-applications/requirement-form.tsx

```diff
@@ -212,6 +212,8 @@ export const RequirementForm = observer(
         // formio `pristine` is not set for file upldates
         // using `setPristine(false)` causes the entire form to validate so instead, we use a separate dirty state
         setIsDirty(true)
+        // trigger save to rerun compliance and save file
+        triggerSave?.({ autosave: true, skipDirtyCheck: true })
```

Comment:
@LaurelOlson yeah I tried running the DigitalSealValidator in the permit application upload endpoint but it doesn't push the updated compliance result and gets stale data. I don't have too much context on the feature so I am not sure why but I think it's because the submittedFormData has to have the filed for it to run auto compliance on a field. So that's why I run the autosave right after file is uploaded. This shouldn't have any impact either because the form autosaves automatically after 60 seconds, but this just ensures that the result is updated more quickly as that's a long wait time.

Reply:
@taw27 OK sounds good - can you remove the isDirty logic then? It was added specifically for file uploads since they do not trigger the check but since you are adding the `skipDirtyCheck` it removes the need for separately keeping track of `isDirty` and we can just use the `pristine` flag in formIO.